gems/smithy/lib/smithy/welds/endpoints.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Welds
+
+    # Provides default endpoint builtin/function bindings.
+    # TODO : There are many ways we could "register" the base (ie, non-AWS) endpoint builtins/functions
+    # I like this since it uses the same extension point that AWS will and makes it easy to remove/replace.
+    class Endpoints < Weld
+
+      def self.preprocess(model)
+        # TODO: There are quite a few places we could do this
+        # This is one idea - I like this because it keeps all of the "default" endpoint logic in an extension point rather than in core logic
+        # that makes it easy to remove/replace.
+        # add default endpoint rules/tests if none are set
+        model['shapes'].select { |_k, s| s['type'] == 'service' }.each do |_name, shape|
+          unless shape['traits']['smithy.rules#endpointRuleSet']
+            add_default_endpoints(shape['traits'])
+          end
+        end
+      end
+      def self.built_in_bindings
+        [Vise::Endpoints::BuiltInBinding.new(
+          id: 'SDK::Endpoint',
+          render_config: proc do |_plan|
+            <<-ADD_OPTION
```

Comment:
I've been using ~ with indentation. I think you can make this a method that takes option values. Custom options come from the endpoint params but static values for endpoint.

Reply:
I can switch to `<<~` for indentation - I don't have a style preference on those.

What do you mean about having this method take option values.  I'm not quite sure I understand.

In general, this needs to be fairly flexible to support all of our existing/future built ins (eg: account id).

Reply:
This still needs to be indented. What I was saying before was to have a method that is responsible for rending the config, but this is fine to hardcode endpoint for now. Eventually this will need to be a method.

Reply:
This is the correct indenting for it to render correctly in the generated code.

Why should this be a method?  I think I'm still a little confused about that. Using a proc/callable is what lets a user define arbitrary configuration(s) that may apply for the builtin.  I think thats perferable to coming up with a codegen time modeling of config.